services/autorust/codegen/src/codegen_operations.rs

```diff
@@ -646,6 +655,113 @@ struct ResponseCode {
+        let header_name = header_name.to_lowercase(); // ETag as etag
+        let type_name = match (header.type_.as_str(), header.format.as_deref()) {
+            ("string", None) => Ok(TypeName::String),
+            ("string", Some("byte")) => Ok(TypeName::String), // base64-encoded
+            ("string", Some("duration")) => Ok(TypeName::String),
+            ("string", Some("uuid")) => Ok(TypeName::String),
+            ("string", Some("date-time-rfc1123")) => Ok(TypeName::DateTimeRfc1123),
+            ("integer", None) => Ok(TypeName::Int32),
+            ("integer", Some("int32")) => Ok(TypeName::Int32),
+            ("number", None) => Ok(TypeName::Float32),
+            ("number", Some("float")) => Ok(TypeName::Float32),
+            ("number", Some("double")) => Ok(TypeName::Float64),
+            ("integer", Some("int64")) => Ok(TypeName::Int64),
+            ("boolean", None) => Ok(TypeName::Boolean),
+            (header_type, header_format) => Err(Error::with_message(ErrorKind::CodeGen, || {
+                format!("header type '{}' format '{:?}' not matched", header_type, header_format)
+            })),
+        }?;
+        let type_name_code = TypeNameCode::new(&type_name)?;
+        let description = DocCommentCode::new(header.description.clone());
+        Ok(Self {
+            header_name,
+            function_name,
+            type_name_code,
+            description,
+        })
+    }
+}
+
+impl ToTokens for HeaderCode {
+    fn to_tokens(&self, tokens: &mut TokenStream) {
+        let Self {
+            header_name,
+            function_name,
+            type_name_code,
+            description,
+        } = &self;
+        description.to_tokens(tokens);
+        if type_name_code.is_string() {
+            tokens.extend(quote! {
+                pub fn #function_name(&self) -> azure_core::Result<&str> {
+                    self.0.get_str(&azure_core::headers::HeaderName::from_static(#header_name))
+                }
+            });
+        } else if type_name_code.is_date_time() {
+            tokens.extend(quote! {
+                pub fn #function_name(&self) -> azure_core::Result<time::OffsetDateTime> {
+                    azure_core::date::parse_rfc3339(self.0.get_str(&azure_core::headers::HeaderName::from_static(#header_name))?)
+                }
+            });
+        } else if type_name_code.is_date_time_rfc1123() {
+            tokens.extend(quote! {
+                pub fn #function_name(&self) -> azure_core::Result<time::OffsetDateTime> {
+                    azure_core::date::parse_rfc1123(self.0.get_str(&azure_core::headers::HeaderName::from_static(#header_name))?)
+                }
+            });
+        } else {
+            tokens.extend(quote! {
+                pub fn #function_name(&self) -> azure_core::Result<#type_name_code> {
+                    self.0.get_as(&azure_core::headers::HeaderName::from_static(#header_name))
+                }
+            });
+        }
+    }
```

Comment:
It would be clearer if you assigned the `quote!` values to a variable and only call `tokens.extend` once at the end.